thunder/torch/__init__.py

```diff
@@ -1589,6 +1589,37 @@ def where(
     return clang.where(pred, a, b)
 
 
+@torchsymbol(torch.nan_to_num, is_method=True)
+def nan_to_num(
+    a: TensorLike,
+    nan: None | Number = 0.0,
+    posinf: None | Number = None,
+    neginf: None | Number = None,
+    /,
+    out: None | TensorLike = None,
+):
+
+    utils.check(out is None, lambda: "out is not None which is currently unsupported", NotImplementedError)
+
+    if dtypes.is_boolean_dtype(a.dtype) or dtypes.is_integer_dtype(a.dtype):
+        # NOTE PyTorch returns a.clone()
+        return a
+
+    if nan is None:
+        nan = 0.0
+
+    if posinf is None:
+        posinf = torch.finfo(to_torch_dtype(a.dtype)).max
```

Comment:
Do you know what happens in PyTorch if the underlying type is, say, `float`, but a max for `double` is provided?

Reply:
Ah interesting! In such case, `nan_to_num` doesn't replace the values. 
```python
a = torch.randn(3)
a[0] = float("inf")
b = a.double()
b_max = torch.finfo(b.dtype).max
a_out = torch.nan_to_num(a, nan=None, posinf=b_max)
print(a == a_out) # tensor([True, True, True])
```